lib/apphook.c

```diff
@@ -95,15 +95,21 @@ register_application_hook(gint type, ApplicationHookFunc func, gpointer user_dat
 }
 
 static void
-run_application_hook(gint type, gboolean remove_hook)
+_update_current_state(gint type)
+{
+  if (AH_REOPEN == type)
```

Comment:
Why do we need this early return?

Reply:
Because the `AH_REOPEN` is kinda hack here. The app hook has its internal state machine, stepping from daemonize, pre config, post config, and so on. This function updates that.
A real deal would be to untangle the reopen from the rest of the hooks.

Reply:
Oh ok this is indeed an equivalent transformation, just a tricky one: only AH_REOPEN had remove hook false.